### PR TITLE
Implement CompositeEventProcessor

### PR DESCRIPTION
**Pull request by dimitry-**
CompositeEventProcessor is a CaptureEventProcessor that
combines other CaptureEventProcessors.

Test: run CaptureClientTests

### CODE REVIEW

File: src/CaptureClient/CompositeEventProcessor.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureClient/CaptureEventProcessor.h"
+
+namespace orbit_capture_client {
+
+namespace {
+
+class CompositeCaptureEventProcessor : public CaptureEventProcessor {
```
Comment (dpallotti) [inline src/CaptureClient/CompositeEventProcessor.cpp:11]: The name of the file doesn't reflect the name of the class (or vice versa).

File: src/CaptureClient/CompositeEventProcessorTest.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#include "CaptureClient/CaptureEventProcessor.h"
+
+namespace orbit_capture_client {
+
+class MockEventProcessor : public CaptureEventProcessor {
+ public:
+  MOCK_METHOD(void, ProcessEvent, (const orbit_grpc_protos::ClientCaptureEvent& event), (override));
+};
+
+TEST(CompositeEventProcessor, Smoke) {
```
Comment (dpallotti) [inline src/CaptureClient/CompositeEventProcessorTest.cpp:17]: The name of the test doesn't reflect the name of the class (or vice versa).

**Review by dpallotti — COMMENTED** [review]
Just some naming consistency to fix. I would say that both `CompositeCaptureEventProcessor` and `CompositeEventProcessor` would be fine. But if you choose the latter I would advocate for renaming `SaveToFileCaptureEventProcessor` to `SaveToFileEventProcessor` (for consistency).

File: src/CaptureClient/CompositeEventProcessor.cpp (hunk 1 above)
Comment (dimitry-) [reply src/CaptureClient/CompositeEventProcessor.cpp:11]: done.

File: src/CaptureClient/CompositeEventProcessorTest.cpp (hunk 2 above)
Comment (dimitry-) [reply src/CaptureClient/CompositeEventProcessorTest.cpp:17]: done

**Review by florian-kuebler — APPROVED** [review]
LGTM

**Review by dpallotti — APPROVED** [review]

**Pull request merged**